packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.routing.js

```diff
@@ -1,43 +1,103 @@
+import get from 'lodash/get';
+
 export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('app.hosting.cdn.order', {
     url: '/order',
     component: 'hostingCdnOrder',
     resolve: {
-      goBack: /* @ngInject */ goToHosting => goToHosting,
+      autoPayWithPreferredPaymentMethod: /* @ngInject */
+        ovhPaymentMethod => ovhPaymentMethod.hasDefaultPaymentMethod(),
+
+      catalogAddon: /* @ngInject */ (
+        goBack,
+        serviceOption,
+        user,
+        $translate,
+        HostingCdnOrderService,
+      ) => HostingCdnOrderService
+        .getCatalogAddon(user.ovhSubsidiary, serviceOption)
+        .catch(error => goBack(
+          $translate.instant('hosting_dashboard_cdn_order_error', { message: get(error, 'data.message', error) }),
+          'danger',
+        )),
 
-      availableOffers: /* @ngInject */ (
+      checkoutOrderCart: /* @ngInject */ (
+        goBack,
         isOptionFree,
         $translate,
-        HostingOptionOrder,
-      ) => HostingOptionOrder
-        .getOrderEnums('hosting.web.CdnOfferEnum')
-        .then((models) => {
-          const getOfferName = (offer) => {
-            const offerKey = `hosting_dashboard_cdn_order_${offer}`;
-            const offerName = $translate.instant(offerKey);
-
-            return (offerName === offerKey) ? offer : offerName;
-          };
-
-          const availableOffers = !isOptionFree
-            ? models.filter(offer => offer !== 'CDN_BUSINESS_FREE')
-            : ['CDN_BUSINESS_FREE'];
-
-          return availableOffers.map(offer => ({
-            label: getOfferName(offer),
-            value: offer,
-          }));
-        }),
-      hosting: /* @ngInject */ (productId, Hosting) => Hosting.getSelected(productId),
-      hostingProxy: /* @ngInject */ (productId, Hosting) => Hosting.getHosting(productId),
-      productId: /* @ngInject */ $transition$ => $transition$.params().productId,
-
-      isOptionFree: /* @ngInject */ (hosting, isPerfOffer) => isPerfOffer() || hosting.isCloudWeb,
-      isOrderable: /* @ngInject */ (productId, HostingOptionOrder) => HostingOptionOrder
-        .isOptionOrderable('cdn', productId),
-      isPerfOffer: /* @ngInject */ (hostingProxy, Hosting) => () => Hosting
-        .constructor
-        .isPerfOffer(hostingProxy.offer),
+        $window,
+        HostingCdnOrderService,
+      ) => async (
+        autoPayWithPreferredPaymentMethod,
+        cartId,
+      ) => {
+        try {
+          const order = await HostingCdnOrderService
+            .checkoutOrderCart(autoPayWithPreferredPaymentMethod, cartId);
+
+          if (isOptionFree) {
+            goBack(
+              $translate.instant('hosting_dashboard_cdn_order_success_activation'),
+            );
+          } else {
+            $window.open(order.url, '_blank');
+            goBack(
+              $translate.instant('hosting_dashboard_cdn_order_success', { t0: order.url }),
```

Comment:
I wonder if we really need to try to go on this state to then go back on its parent

Reply:
I don't understand your comment. About what, the `try / catch` ?
Following the checkout state, we show the success/error message on the dashboard.
I've kept the behaviour of the refactored page for this one.

Were you talking about showing this message directly on the order page ?

Reply:
I think I might have misunderstood the code, my bad 😅